crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -1622,16 +1623,19 @@ impl<'db> TypeInferenceBuilder<'db> {
             range: _,
             target,
             annotation,
-            value,
+            value: _,
             simple: _,
         } = assignment;
 
-        let mut annotation_ty = self.infer_annotation_expression(annotation);
+        let mut annotation_ty = self.infer_annotation_expression(
+            annotation,
+            DeferredState::from(self.are_all_types_deferred()),
+        );
 
         // Handle various singletons.
         if let Type::Instance(InstanceType { class }) = annotation_ty {
             if class.is_known(self.db, KnownClass::SpecialForm) {
-                if let Some(name_expr) = target.as_name_expr() {
+                if let Some(name_expr) = assignment.target.as_name_expr() {
```

Comment:
Why the change to use `assignment.target` here instead of the `target` we unpacked above? Shouldn't they be the same?

Reply:
Sorry, this is just a leftover from my initial attempt, will revert